### PR TITLE
New analyser: fix crashes when plugins encounter a placeholder node

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Fixes https://github.com/python/mypy/issues/7098

### CODE REVIEW

File: mypy/plugins/attrs.py [hunk 1]
```diff
@@ -270,7 +270,9 @@ def _analyze_class(ctx: 'mypy.plugin.ClassDefContext',
         # instance level assignments.
         if attribute.name in ctx.cls.info.names:
             node = ctx.cls.info.names[attribute.name].node
-            assert isinstance(node, Var)
+            if not isinstance(node, Var):
+                # This node may be not ready yet (i.e. a PlaceholderNode).
```
Comment (JukkaL) [inline mypy/plugins/attrs.py]: Should we instead check for `PlaceholderNode`? If there's something unexpected there, weird things might happen.

File: test-data/unit/check-attr.test [hunk 2]
```diff
@@ -1107,3 +1107,47 @@ C(43)
...
+
+[case testTypeInAttrBad]
+import attr
+
+@attr.s
+class C:
+    total = attr.ib(type=Bad)  # E: Name 'Bad' is not defined
+[builtins fixtures/bool.pyi]
+
+[case testTypeInAttrBad2]
```
Comment (JukkaL) [inline test-data/unit/check-attr.test]: Maybe rename the test case? There seems nothing bad here, just a forward reference.

File: test-data/unit/check-attr.test [hunk 3]
```diff
@@ -1107,3 +1107,47 @@ C(43)
...
+@attr.s
+class C:
+    total = attr.ib(type=Forward)
+
+reveal_type(C.total)  # N: Revealed type is '__main__.Forward'
+C('no')  # E: Argument 1 to "C" has incompatible type "str"; expected "Forward"
+class Forward: ...
+[builtins fixtures/bool.pyi]
+
+[case testTypeInAttrBad3]
```
Comment (JukkaL) [inline test-data/unit/check-attr.test]: Similar to above, the test case name doesn't seem quite right.

**Comment by ilevkivskyi** [comment]
Travis looks stuck. I am going to merge this now.

**Pull request merged**